### DISCUSSION TITLE
I like have `w-4` and `w-4-em` at the same time

### DISCUSSION THREAD

**Discussion by babakfp** [tailwindlabs/tailwindcss#9941] [Help] [answered]
Hi

I tried using this https://tailwindcss.com/docs/theme#referencing-other-values but it throws this error `[postcss] Maximum call stack size exceeded`.

I'm using a plugin for tailwindcss and in that plugin, there is something like this https://babakfp.ir/docs/tailwindcss-addons/presets, so I'm going to need `w-4-em` to work with all of the spacing values and `min-width`, `max-width` and etc.

Thanks.

**Comment by wongjn** [comment, marked as answer]
Are you possibly trying to do something like the following?
```js
/** @type {import('tailwindcss').Config} */
module.exports = {
  theme: {
    extend: {
      width: ({ theme }) => ({
        foo: theme('width.4'),
      }),
    },
  },
};
```
This would indeed throw the error you describe, since the way that Tailwind configuration resolves, your function would keep calling itself. You could try using the `spacing` key in the `theme()` call instead, since many of the `width` values come from this key by default:
```diff
 /** @type {import('tailwindcss').Config} */
 module.exports = {
   theme: {
     extend: {
       width: ({ theme }) => ({
-        foo: theme('width.4'),
+        foo: theme('spacing.4'),
       }),
     },
   },
 };
```
Or otherwise, iterate with your own values without using `theme()` at all.
